app/models/concerns/portered.rb

```diff
@@ -0,0 +1,79 @@
+# encoding: utf-8
+
+require 'aws-sdk'
+
+module Portered
+  extend ActiveSupport::Concern
+
+  unless ENV['PORTER_SNS_TOPIC_ARN'] || Rails.env.test?
+    Rails.logger.warn('No Porter SNS topic provided - Porter jobs will be skipped.')
+  end
+
+  class_methods do
+    def porter_callbacks(callbacks = nil)
+      if callbacks.present?
+        @porter_callbacks = Array.wrap(callbacks).map { |callback| format_callback(callback) }
+      end
+      @porter_callbacks
+    end
+
+    def format_callback(callback)
+      if callback[:sqs].present?
+        {
+          Type: 'AWS/SQS',
+          Queue: URI::HTTPS.build(
+            host: "sqs.#{ENV['AWS_REGION']}.amazonaws.com",
+            path: "/#{ENV['AWS_ACCOUNT_ID']}/#{callback[:sqs]}"
+          ).to_s
+        }
+      elsif callback[:s3].present?
```

Comment:
Nice - you're using this for testing, i presume?

Reply:
yep!